### PR TITLE
fix(routing): stopsKey option in RoutingControl

### PR DESCRIPTION
**Pull request by danji90**
# How to

Currently there is no way to control the key used for station lookups (when station uid is passed instead of coordinates).
An option _stopsKey_ was added, which controls the api key used for station lookups.

Additionally the image was replaced for the RoutingControl card in the examples in https://mobility-toolbox-js.geops.io/examples

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [x] The new class' members & methods are well documented
- [x] Tests added.

### CODE REVIEW

File: src/ol/controls/RoutingControl.js [hunk 1]
```diff
@@ -65,7 +65,9 @@ class RoutingControl extends Control {
 
     this.abortController = new AbortController();
 
+    // By default one apiKey is used. Optionally the api key for the tops lookup can be configured
```
Comment (friedjoff) [inline src/ol/controls/RoutingControl.js:68]:
```suggestion
    // By default one apiKey is used. Optionally the api key for the stops lookup can be configured
```

File: src/ol/controls/RoutingControl.js [hunk 2]
```diff
@@ -65,7 +65,9 @@ class RoutingControl extends Control {
...
     this.apiKey = options.apiKey;
+    this.stopsKey = options.stopsKey || this.apiKey;
```
Comment (friedjoff) [inline src/ol/controls/RoutingControl.js:70]:
```suggestion
    this.stopsApiKey = options.stopsApiKey || this.apiKey;
```

File: src/ol/controls/RoutingControl.js [hunk 3]
```diff
@@ -275,6 +248,33 @@ class RoutingControl extends Control {
...
 
+  drawViaPoint(viaPoint, idx) {
+    const pointFeature = new Feature();
+    pointFeature.set('viaPointIdx', idx);
+    if (Array.isArray(viaPoint)) {
+      pointFeature.setGeometry(new Point(viaPoint));
+      return this.routingLayer.olLayer.getSource().addFeature(pointFeature);
+    }
+    return fetch(
+      `https://api.geops.io/stops/v1/lookup/${viaPoint}?key=${this.stopsKey}`,
```
Comment (friedjoff) [inline src/ol/controls/RoutingControl.js:259]: Shouldn't the stops API URL also be configurable?

**Review by friedjoff — CHANGES_REQUESTED** [review]
Please also add JSDoc for all options.

File: src/ol/controls/RoutingControl.js (hunk 3 above)
Comment (danji90) [reply src/ol/controls/RoutingControl.js]: Guess that makes sense

File: src/ol/controls/RoutingControl.js (hunk 1 above)
Comment (danji90) [reply src/ol/controls/RoutingControl.js]: done

File: src/ol/controls/RoutingControl.js [hunk 4]
```diff
@@ -61,24 +66,35 @@ class RoutingControl extends Control {
...
+    /** @ignore */
     this.apiKey = options.apiKey;
 
+    /** @ignore */
+    this.stopsApiKey = options.stopsApiKey || this.apiKey;
+
+    /** @ignore */
+    this.stopsApiUrl =
+      options.stopsApiUrl || 'https://api.geops.io/stops/v1/lookup/';
+
```
Comment (oterral) [inline src/ol/controls/RoutingControl.js:83]: Should the lookup url use the this.url property ?

Comment (danji90) [reply src/ol/controls/RoutingControl.js:83]: yes. Maybe the  name for the variable is a bit confusing, since we don's actually use the StopsAPI class

**Review by oterral — APPROVED** [review]
lgtm

**Pull request merged**